conntrack.go

```diff
@@ -380,7 +381,44 @@ func (nfct *Nfct) register(ctx context.Context, t Table, groups NetlinkGroup, fi
 					nfct.logger.Printf("could not parse received message: %v", err)
 					continue
 				}
-				if ret := fn(c); ret != 0 {
+				// Attempt to detect the table of the received event
+				var nlTable Table
+				msgTypeTable := (msg.Header.Type&0x300)>>8
+				switch msgTypeTable {
+				case unix.NFNL_SUBSYS_CTNETLINK:
+					nlTable = Conntrack
+				case unix.NFNL_SUBSYS_CTNETLINK_EXP:
+					nlTable = Expected
+				default:
+					nfct.logger.Printf("Unable to detect the table of the received message. Unknown table: %v", msgTypeTable)
+				}
+				// Attempt to detect the group of the received event
+				var nlGroup NetlinkGroup
+				msgTypeGroup := (msg.Header.Type&0xF)
```

Comment:
Similar to the comment on the table type. For performance reasons I would also suggest to directly pass this value to the user instead of adding more complexity.

Reply:
Still, for performance reasons I would prefer passing the value directly.

If a user is interested in more details a helper function like `Is(group, target NetlinkGroup, table Table) bool` should be fine.